File: 1653-minimum-deletions-to-make-string-balanced/1653-minimum-deletions-to-make-string-balanced.cpp
class Solution {
public:
    int minimumDeletions(string s) {
        int res = 0;
        int n = s.length();
        int a = 0;
        vector<int> suffB(n, 0);
        for(int i = n - 1; i >= 0; i--)
        {
            if(s[i] == 'b')
            {
                suffB[i]++;
            }
            if(i < n - 1)
            {
                suffB[i] += suffB[i + 1];
            }
        }
        for(int i = 0; i < n; i++)
        {
            if(s[i] == 'a')
            {
                a++;
            }
            res = max(res, a + suffB[i]);
        }
        return n - res;
    }
};

File: 1653-minimum-deletions-to-make-string-balanced/1653-minimum-deletions-to-make-string-balanced.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        res = 0
        a = 0
        suffB = [0 for _ in range(len(s))]
        for i in range(len(s) - 1, -1, -1):
            if s[i] == 'b':
                suffB[i] += 1
            if i < len(s) - 1:
                suffB[i] += suffB[i + 1]
        for i in range(len(s)):
            if s[i] == 'a':
                a += 1
            res = max(res, a + suffB[i])
        return len(s) - res